app/src/app/[lng]/[inventory]/layout.tsx

```diff
@@ -24,10 +24,15 @@ export default function DataLayout({
   const { data: userInfo, isLoading: isUserInfoLoading } =
     api.useGetUserInfoQuery();
 
-  const inventoryId = userInfo?.defaultInventoryId || inventory;
+  let inventoryId: string | null;
+  if (inventory && inventory !== "null") {
+    inventoryId = inventory;
+  } else {
+    inventoryId = userInfo?.defaultInventoryId ?? null;
+  }
 
   const { data: inventoryOrgData, isLoading: isInventoryOrgDataLoading } =
-    useGetOrganizationForInventoryQuery(inventoryId, {
+    useGetOrganizationForInventoryQuery(inventoryId!, {
```

Comment:
by the logic on line 31, inventoryId can be null :confused:

Reply:
Yes, this gives a type error without the `!`. It will never be passed into the API as when it's null the skip condition will be active.